staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go

```diff
@@ -46,52 +49,114 @@ func init() {
 // featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.<FeatureName>, true)
 func SetFeatureGateDuringTest(tb TB, gate featuregate.FeatureGate, f featuregate.Feature, value bool) {
 	tb.Helper()
-	detectParallelOverrideCleanup := detectParallelOverride(tb, f)
-	originalValue := gate.Enabled(f)
+	SetFeatureGatesDuringTest(tb, gate, FeatureOverrides{f: value})
+}
+
+// SetFeatureGatesDuringTest sets the specified map of feature gate values for duration of the test.
+// Fails when it detects second call to the same flag or is unable to set or restore feature flag.
+// When disabling a feature, this automatically disables all dependents that weren't explicitly set.
+//
+// WARNING: Can leak set variable when called in test calling t.Parallel(), however second attempt to set the same feature flag will cause fatal.
+//
+// Example use:
+//
+//	featuregatetesting.SetFeatureGatesDuringTest(t, utilfeature.DefaultFeatureGate, featuregatetesting.FeatureOverrides{
+//		features.<FeatureName>: true,
+//	    features.<FeatureName>: false,
+//	})
+func SetFeatureGatesDuringTest(tb TB, gate featuregate.FeatureGate, features FeatureOverrides) {
+	tb.Helper()
 	originalEmuVer := gate.(featuregate.MutableVersionedFeatureGate).EmulationVersion()
-	originalExplicitlySet := gate.(featuregate.MutableVersionedFeatureGate).ExplicitlySet(f)
+	originalValues := map[string]bool{}
+	var originalUnset []featuregate.Feature
+	overrides := FeatureOverrides{}
+
+	// Automatically disable dependents when disabling a dependency.
+	dependencies := gate.Dependencies()
+	for f := range dependencies {
+		if gate.Enabled(f) && hasDisabledDependency(f, dependencies, features) {
+			overrides[f] = false
+		}
+	}
 
 	// Specially handle AllAlpha and AllBeta
-	if f == "AllAlpha" || f == "AllBeta" {
+	allAlphaValue, allAlpha := features["AllAlpha"]
+	allBetaValue, allBeta := features["AllBeta"]
+	if allAlpha || allBeta {
 		// Iterate over individual gates so their individual values get restored
 		for k, v := range gate.(featuregate.MutableFeatureGate).GetAll() {
-			if k == "AllAlpha" || k == "AllBeta" {
```

Comment:
are `AllAlpha` and `AllBeta` not included in the `GetAll` response in ways we have to skip?